### PR TITLE
create repository

### PR DESCRIPTION
**Pull request by nguyenduccanh2501**

### CODE REVIEW

**Comment by nguyenduccanh2501** [comment]
ready

File: app/src/main/java/com/sun/moviedb_45/data/source/MovieRepository.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import com.sun.moviedb_45.data.source.local.MovieLocalDataSource;
+import com.sun.moviedb_45.data.source.remote.MovieRemoteDataSource;
+import com.sun.moviedb_45.service.response.MovieResponse;
+
+import io.reactivex.Observable;
+
+public class MovieRepository implements MovieDataSource.Local, MovieDataSource.Remote {
+    private static MovieRepository sInstance;
+    private MovieRemoteDataSource mRemoteDataSource;
+    private MovieLocalDataSource mLocalDataSource;
```
Comment (tuanbda-0832) [inline app/src/main/java/com/sun/moviedb_45/data/source/MovieRepository.java]:
```suggestion
    private MovieDataSource.Local mLocalDataSource;
```
the high-level module must never rely on any low-level module but should depend on abstractions

File: app/src/main/java/com/sun/moviedb_45/data/source/MovieRepository.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.sun.moviedb_45.data.source.local.MovieLocalDataSource;
+import com.sun.moviedb_45.data.source.remote.MovieRemoteDataSource;
+import com.sun.moviedb_45.service.response.MovieResponse;
+
+import io.reactivex.Observable;
+
+public class MovieRepository implements MovieDataSource.Local, MovieDataSource.Remote {
+    private static MovieRepository sInstance;
+    private MovieRemoteDataSource mRemoteDataSource;
```
Comment (tuanbda-0832) [inline app/src/main/java/com/sun/moviedb_45/data/source/MovieRepository.java]: similar line 12

File: app/src/main/java/com/sun/moviedb_45/data/source/MovieRepository.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private MovieLocalDataSource mLocalDataSource;
+
+    private MovieRepository(MovieRemoteDataSource remote,
```
Comment (tuanbda-0832) [inline app/src/main/java/com/sun/moviedb_45/data/source/MovieRepository.java]:
```suggestion
    private MovieRepository(MovieDataSource.Remote remote,
```

File: app/src/main/java/com/sun/moviedb_45/data/source/MovieRepository.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                            MovieLocalDataSource local) {
```
Comment (tuanbda-0832) [inline app/src/main/java/com/sun/moviedb_45/data/source/MovieRepository.java]:
```suggestion
                            MovieDataSource.Local local) {
```

**Pull request merged**